cocos2d/core/event/event-target.js

```diff
@@ -34,20 +34,22 @@ var _doDispatchEvent = function (owner, event) {
     event.target = owner;
 
     // Event.CAPTURING_PHASE
-    cachedArray.length = 0;
-    owner._getCapturingTargets(event.type, cachedArray);
-    // capturing
-    event.eventPhase = 1;
-    for (i = cachedArray.length - 1; i >= 0; --i) {
-        target = cachedArray[i];
-        if (target._isTargetActive(event.type) && target._capturingListeners) {
-            event.currentTarget = target;
-            // fire event
-            target._capturingListeners.invoke(event);
-            // check if propagation stopped
-            if (event._propagationStopped) {
-                cachedArray.length = 0;
-                return;
+    if (!event._propagationStopped) {
```

Comment:
If propagation is stopped from the beginning, event will still send to target itself